extensions/who-is-visiting-this-content/app.R

```diff
@@ -177,38 +314,145 @@ server <- function(input, output, session) {
 
     filtered_visits |>
       left_join(unfiltered_hits, by = "user_guid") |>
-      left_join(user_names(), by = "user_guid") |>
-      replace_na(list(full_name = "[Anonymous]")) |>
+      left_join(users(), by = "user_guid") |>
+      replace_na(list(display_name = "[Anonymous]")) |>
       arrange(desc(n_visits)) |>
-      select(n_visits, n_hits, full_name, username)
+      select(user_guid, display_name, email, n_visits)
   })
 
-  summary_message <- reactive({
-    content_title <- content() |>
-      filter(guid == input$content_guid) |>
-      pull(title)
+  selected_content_info <- reactive({
+    filter(content(), guid == input$guid_field)
+  })
+
+  # Create day by day hit data for plot
+  daily_hit_data <- reactive({
+    all_dates <- seq.Date(date_range()[1], date_range()[2], by = "day")
+
+    all_visits_data() |>
+      mutate(date = date(timestamp)) |>
+      group_by(date) |>
+      summarize(daily_visits = n(), .groups = "drop") |>
+      tidyr::complete(date = all_dates, fill = list(daily_visits = 0))
+  })
+
+  # Render tabular output ----
+
+  output$aggregated_visits <- renderReactable({
+    reactable(
+      aggregated_visits_data(),
+      selection = "single",
+      onClick = "select",
+      defaultSorted = "n_visits",
+      columns = list(
+        user_guid = colDef(show = FALSE),
+        display_name = colDef(name = "Visitor"),
+        email = colDef(
+          name = "",
+          width = 32,
+          sortable = FALSE,
+          cell = function(url) {
+            if (is.na(url) || url == "") return("")
+            HTML(as.character(tags$div(
+              onclick = "event.stopPropagation()",
+              tags$a(
+                href = glue("mailto:{url}"),
+                bs_icon("envelope")
+              )
+            )))
+          },
+          html = TRUE
+        ),
+        n_visits = colDef(
+          name = "Visits",
+          defaultSortOrder = "desc",
+          maxWidth = 75
+        )
+      )
+    )
+  })
+
+  output$all_visits <- renderReactable({
+    reactable(
+      all_visits_data(),
+      defaultSorted = "timestamp",
+      columns = list(
+        user_guid = colDef(show = FALSE),
+        timestamp = colDef(
+          name = "Time",
+          format = colFormat(datetime = TRUE, time = TRUE),
+          defaultSortOrder = "desc"
+        ),
+        display_name = colDef(name = "Visitor")
+      )
+    )
+  })
+
+  # Render content metadata and other text ----
+
+  output$filter_message <- renderUI({
+    # req(getReactableState("aggregated_visits", "selected"))
     hits <- all_visits_data()
     glue(
-      "Content '{content_title}' had {nrow(hits)} visits between ",
-      "{min(hits$timestamp)} and {max(hits$timestamp)}."
+      "{nrow(hits)} visits between ",
+      "{date_range()[1]} and {date_range()[2]}."
```

Comment:
Same comment here about `from_date` and `to_date` it's _pretty obvious_ but if we have the names, that would be lovely to use